src/solver.cpp

```diff
@@ -36,10 +36,20 @@ Solver::Solver(const Solver &other)
       maxstepsB_(other.maxstepsB_), sensi_(other.sensi_)
 {}
 
+void Solver::apply_max_num_steps() const {
+    // set remaining steps, setMaxNumSteps only applies to a single call of solve
+    long int cursteps;
+    getNumSteps(solver_memory_.get(), &cursteps);
+    setMaxNumSteps(maxsteps_-cursteps);
```

Comment:
Wait, we need to be careful here: in the (probably rare) case, where the solver just reaches `tout` with `maxsteps` steps, this passes `0` to SUNDIALS, which is interpreted as default value (500). In the following step, this will get negative which is interpreted as "no limit".

Reply:
Corner-case, but that would be fun to debug :D

Reply:
good catch!